tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/nvptx_backend_lib.cc

```diff
@@ -319,6 +354,36 @@ StatusOr<string> CompileModuleToPtx(llvm::Module* module,
     }
   }
 
+  return Status::OK();
+}
+
+std::unique_ptr<llvm::TargetMachine> NVPTXGetTargetMachine(
+    llvm::Triple target_triple, GpuVersion gpu_version,
+    const HloModuleConfig& hlo_module_config) {
+  // Figure out the exact name of the processor as known to the NVPTX backend
+  // from the gpu_architecture flag.
+  return GetTargetMachine(
+      target_triple, GetSmName(absl::get<std::pair<int, int>>(gpu_version)),
```

Comment:
Technically an AMDGPU GpuVersion could have been passed by the caller. Please handle this error case. Same for the AMDGPU-specific function. Alternatively, do not use the GpuVersion abstraction, so this error case can never happen.

Reply:
let me change `NVPTXGetTargetMachine` to take `std::pair<int, int>` and handle the case in `CompileToPtx`.